### PR TITLE
confgenerator: change otel self-metric port

### LINKED ISSUE #461: Add support to customize port number for agent metrics
Any chance we will be able to change the port or change it to something that is not conflicting?
https://github.com/GoogleCloudPlatform/ops-agent/blob/master/confgenerator/agentmetrics.go#L36

As one of the Druid components uses port 8888 and its conflicting with ops-agent

`Mar 10 13:08:21 otelopscol: 2022-03-10T13:08:21.717Z#011error#011service/collector.go:153#011Asynchronous error received, terminating process#011{"error": "listen tcp :8888: bind: address already in use"}`

### PR DESCRIPTION
**Pull request by braydonk**
Port 8888 is popular with a number of applications such as Tomcat,
Jetty, Apache Druid, etc. Until we finish designing dynamic port choice,
this PR will change port 8888 to something far less common. I've opted
to change it to 20201, to be closer to the fluent-bit default of 20202.

[EDIT] Not quite enough for #461 — that asks for user-level customization.

### CODE REVIEW

**Review by igorpeshansky — APPROVED** [review]
LGTM :shipit:

File: confgenerator/otel/modular.go [hunk 1]
```diff
@@ -22,6 +22,8 @@ import (
 	"github.com/mitchellh/mapstructure"
 )
 
+var MetricsPort = 20201
```
Comment (jefferbrecht) [inline confgenerator/otel/modular.go:25]: This can be const.

**Review by jefferbrecht — APPROVED** [review]
LGTM modulo the const stuff and tests passing 👍  You can add the kokoro:force-run label if it doesn't kick off automatically.

File: confgenerator/otel/modular.go [hunk 2]
```diff
@@ -22,6 +22,8 @@ import (
 	"github.com/mitchellh/mapstructure"
 )
 
+const MetricsPort = 20201
```
[the remark this answers is not in the document]
Comment (braydonk) [reply confgenerator/otel/modular.go:25]: This is exported from the package so it needs to be capitalized. It's used in `agentmetrics.go`

File: confgenerator/fluentbit/metrics.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fluentbit
+
+import "strconv"
+
+const MetricsPort = 20202
```
[the remark this answers is not in the document]
Comment (braydonk) [reply confgenerator/fluentbit/metrics.go:19]: Same as above

**Comment by ravwojdyla** [comment]
Thanks @sophieyfang @braydonk , https://github.com/GoogleCloudPlatform/opentelemetry-operations-collector/issues/86 can probably be closed? How long until this change goes into effect? Also will you update preexisting agent installations?

**Comment by braydonk** [comment]
Hi @ravwojdyla we will be doing a bugfix release 2.12.1 as soon as possible. I don't think we'll be backporting this to anything else so you will need to pick up version 2.12.1 once it's released

**Pull request merged**